src/pages/VenuePage/VenuePage.tsx

```diff
@@ -252,6 +257,31 @@ const VenuePage = () => {
     }
   }, [user, profile, venueId, venueTemplate, mixpanel]);
 
+  // Need a token to access the venue
+  useEffect(() => {
+    // @debt convert this so token is needed to get venue config
+    if (notEmpty(venue.access)) {
+      const token = localStorage.getItem(localStorageTokenKey(venueId));
+      if (!token) {
+        firebase
+          .auth()
+          .signOut()
+          .then(() => setAccessDenied(true));
+      }
+      try {
+        firebase.functions().httpsCallable("access-checkAccess")({
+          venueId,
+          token,
+        });
+      } catch (err) {
+        firebase
+          .auth()
+          .signOut()
+          .then(() => setAccessDenied(true));
```

Comment:
Maybe remove the token from the storage in this case?

Reply:
Good catch!